src/logic/types.ts

```diff
@@ -0,0 +1,28 @@
+enum GridElementState {
+  Hidden,
+  Discovered,
+}
+
+export interface GridElement {
+  state: GridElementState;
+}
+
+export interface GameState {
+  status: GameStatus;
+  board: GridElement[];
+}
+
+export interface GameAction<PayloadType> {
+  type: GameActionType;
+  payload?: PayloadType;
+}
+
+export enum GameStatus {
+  Initial,
+  Running,
+  Ended,
+}
+
+export enum GameActionType {
+  StartGame,
+}
```

Comment:
I think the types should go with their physical declarations in their own files.

So `GameActionType` should go with the actions, `GameState` goes with the `initialState`/reducer and so on and so forth. I think something like `GameStatus` could be in the state but could be here since its more generic.